### PR TITLE
fix(android/app): Handle keyman protocol from external browser

### LINKED ISSUE #4247: bug(android/app): Install link does not work on keyman.com/amharic
**Describe the bug**

Attempting to install the Keyman Amharic keyboard does not work from https://keyman.com/amharic or https://keyman-staging.com/amharic

**To Reproduce**

1. Go to either url above
2. Click on 'Download for Android'
3. Select 'Yes - Install Keyboard'
4. Keyman for Android opens, but no other action is performed

**Expected behavior**

Keyman for Android should start downloading the keyboard immediately.

I have not assessed whether this is an issue with the app or with the website. When resolving this, we should consider backwards compatibility with Keyman 10-13 as well.

---

**Keyman for iPhone/iPad/Android:**
 - Device: Galaxy A90 5G 
 - OS: Android 10
 - Keyman version: 14.0.219-beta

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #4247 and reverts some of a0d1cb0307891fa469064a62228f6450773263c0

* This updates the app to handle `keyman://` links from the internal and external (Chrome) browser.
* This involves converting the legacy `keyman://localhost` link to a kmp install link.

## User Testing
Follow the steps from the reported issue:

> Attempting to install the Keyman Amharic keyboard does not work from https://keyman.com/amharic or https://keyman-staging.com/amharic

> To Reproduce
>
>Go to either url above
>Click on 'Download for Android'
>Select 'Yes - Install Keyboard'

- [ ] Verify Keyman for Android downloads and installs the keyboard package.

### CODE REVIEW

**Review by jahorton — COMMENTED** [review]
I just did a test run and ran into an issue.  What I did:

1.  Compiled and installed the Keyman app.  (Through Android Studio's attached emulation setup.)
2.  Ensure that Get Started displays on app start.  (It may be necessary to disable Keyman from being the system keyboard.)
3. Close / exit the app.
4. Follow the link & sequence from the description.

(After that, I may have attempted a few touch inputs due to unresponsiveness in an attempt to 'rouse' the device.)

What happened:  I ended up with an extremely laggy app launch; I actually thought it was locked, at first.  After a while, though... the Get Started menu was displayed _over_ the package installer UI.  Neither was interactive, but that was because apparently the app had already crashed by the time I got to this point... though the crash hadn't dropped the app's visual elements or returned to the main system screen yet.

**Comment by darcywong00** [comment]
> I just did a test run and ran into an issue. What I did:

> 1. Compiled and installed the Keyman app. (Through Android Studio's attached emulation setup.)
> 2. Ensure that Get Started displays on app start. (It may be necessary to disable Keyman from being the system keyboard.)
> 3. Close / exit the app.
> 4. Follow the link & sequence from the description.

I reran your steps to try to repro
But I'm able to install the keyboard package

gff_amharic downloading (behind Getting Started)
dl behind getting started

I then do the keyboard installation, and the keyboard shows behind Getting Started
amharic in bg

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/util/KMPLink.java [hunk 1]
```diff
@@ -35,6 +35,9 @@
     KMP_STAGING_HOST);
   private static final Pattern downloadPattern = Pattern.compile(downloadPatternFormatStr);
 
+  private static final String LEGACY_DOWNLOAD_KEYBOARDS_PATTERN_FORMATSTR = "^keyman://localhost/open\\?(.+)?$";
+  private static final Pattern legacyDownloadPattermPattern = Pattern.compile(LEGACY_DOWNLOAD_KEYBOARDS_PATTERN_FORMATSTR);
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/util/KMPLink.java]: argh - copy pasta.  Fixed

**Review by jahorton — APPROVED** [review]
Looks like that may have been a one-off.  I don't know when I last launched the VM for that device, so it's possible that my issues were related to that.

Tested again, and things worked fine this time.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.226-beta](https://keyman.com/downloads/releases/beta/14.0.226)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.9-alpha](https://keyman.com/downloads/releases/alpha/15.0.9)

**Pull request merged**